### PR TITLE
web: remove `.theme-redesign` selector from components

### PR DESCRIPTION
**Pull request by 5h1rU**
## Changes

- Removed `.theme-redesign` usage from components.

Part of https://github.com/sourcegraph/sourcegraph/issues/20847.

### CODE REVIEW

File: client/web/src/components/ConnectionPopover.scss [hunk 1]
```diff
@@ -6,50 +6,31 @@
...
         flex-direction: column;
 
-        min-height: 27rem; // avoid jitter when only loading indicator is shown
-        max-height: 70vh;
-
-        .theme-redesign & {
-            min-height: initial;
-            max-height: calc(min(20rem, 70vh));
-        }
+        min-height: initial;
```
Comment (limitedmage) [inline client/web/src/components/ConnectionPopover.scss:18]:
```suggestion
```

File: client/web/src/components/ConnectionPopover.scss [hunk 2]
```diff
@@ -6,50 +6,31 @@
...
 
         .alert {
             word-break: break-word;
         }
     }
 
     &__nodes {
-        &:empty {
-            display: none;
-        }
```
Comment (tjkandala) [inline client/web/src/components/ConnectionPopover.scss:39]: I'm not familiar with this component, but it looks like this didn't change in the redesign, right?

File: client/web/src/components/diff/FileDiffNode.scss [hunk 3]
```diff
@@ -39,14 +33,9 @@
...
-            font-family: $code-font-family;
-            color: var(--body-color);
-            font-size: $code-font-size;
-            .theme-redesign & {
-                color: var(--link-color);
-                font-family: inherit;
-                font-size: 0.875rem;
-            }
+            color: var(--link-color);
+            font-family: inherit;
```
Comment (limitedmage) [inline client/web/src/components/diff/FileDiffNode.scss:37]:
```suggestion
```

**Review by tjkandala — APPROVED** [review]

File: client/web/src/components/ConnectionPopover.scss [hunk 4]
```diff
@@ -59,34 +40,28 @@
...
-                }
+            &:hover,
+            &:focus {
+                background-color: var(--primary);
+                color: var(--light-text);
             }
 
+            &-icon {
+                display: none;
+            }
```
Comment (limitedmage) [inline client/web/src/components/ConnectionPopover.scss:54]: Why was this added?

**Review by limitedmage — APPROVED** [review]

File: client/web/src/components/ConnectionPopover.scss [hunk 5]
```diff
@@ -6,27 +6,15 @@
...
-    --popover-item-padding-v: 0.325rem;
-
-    .theme-redesign & {
-        --popover-item-padding-h: 1rem;
-        --popover-item-padding-v: 0.25rem;
-        background-color: var(--color-bg-1);
-    }
+    --popover-item-padding-h: 1rem;
+    --popover-item-padding-v: 0.25rem;
+    background-color: var(--color-bg-1);
```
Comment (umpox) [inline client/web/src/components/ConnectionPopover.scss:11]: It looks like this isn't being set, it seems to be overridden by the data-reach-tab background color:

image

Do we need to remove `.theme-redesign` from the tabs before merging branch?

File: client/web/src/components/ConnectionPopover.scss [hunk 6]
```diff
@@ -37,19 +25,13 @@
...
             display: none;
         }
-        .theme-redesign & {
-            border-top: solid 1px var(--border-color-2);
-            padding-top: 0.25rem;
-            padding-bottom: 0.25rem;
-        }
+        border-top: solid 1px var(--border-color-2);
+        padding-top: 0.25rem;
+        padding-bottom: 0.25rem;
```
Comment (umpox) [inline client/web/src/components/ConnectionPopover.scss:30]: FYI, this is also overwritten by `FilteredConnection`, a quick fix would be this:

```suggestion
        padding-top: 0.25rem !important;
        padding-bottom: 0.25rem !important;
```

Unless there's a better approach, these classes are used in a few places so not fully confident on a reliable way to increase specificity here. Maybe add an element selector on `ul`?

File: client/web/src/components/ConnectionPopover.scss [hunk 7]
```diff
@@ -59,34 +41,25 @@
             align-items: center;
             padding: var(--popover-item-padding-v) var(--popover-item-padding-h);
             color: var(--text-muted);
+            text-decoration: none;
+            border: none;
```
Comment (umpox) [inline client/web/src/components/ConnectionPopover.scss:45]: This also seems to be overwritten by GitReference:

image

Quick fix:

```suggestion
            border: none !important;
```

Or increase specificity in a better way

**Review by umpox — COMMENTED** [review]
Hey @5h1rU, I noticed some visual regressions in Chromatic - looks like there's some conflicting styles here with a few other components

**Comment by umpox** [comment]
A fix to the clashes with `FilteredConnection` might be to wrap all of the FilteredConnection overrides in `filtered-connection & {`

e.g.

```
    .filtered-connection & {
        &__nodes {
            &:empty {
                display: none;
            }
            border-top: solid 1px var(--border-color-2);
            padding-top: 0.25rem;
            padding-bottom: 0.25rem;
        }

        &__node {
            border-bottom: none;

            &:last-child {
                border-bottom: none;
            }
            &-link {
                display: flex;
                align-items: center;
                padding: var(--popover-item-padding-v) var(--popover-item-padding-h);
                color: var(--text-muted);
                text-decoration: none;
                border: none;

                &:hover,
                &:focus {
                    background-color: var(--primary);
                    color: var(--light-text);
                }

                &--active {
                    font-weight: inherit;
                    background-color: var(--color-bg-3);
                }
            }
        }

        &__input {
            margin: 0.5rem;
            padding-left: 0.5rem;
            padding-right: 0.5rem;
        }
    }
```

These CSS rules are only every used within filtered-connection AFAIK, and we'll always want them to take precedence
